src/Task/TaskMetaData.php

```diff
@@ -38,7 +38,7 @@ public function getKey () : string
 		// these are validated to be safe, so we can keep using these
 		if (null !== $this->uniqueTaskId)
 		{
-			return $this->uniqueTaskId;
+			return strtr($this->uniqueTaskId, ".", "-");
```

Comment:
```suggestion
			return strtr($this->uniqueTaskId, ".", "_");
```

We already use `-` regularly in the name (and `_` practically never, due to CS), so let's use `_` to keep it as a separator